blocks/cards/cards.js

```diff
@@ -29,12 +29,17 @@ async function localizedTopic(topic) {
   return topicResponse[getLocale()] || topicResponse['en-US'] || topicResponse.identifier || topic;
 }
 
+function validPostImage(image) {
+  if (image.startsWith(window.origin)) return image;
+  return (image.charAt(0) === '.') ? window.origin + image.substring(1) : window.origin + image;
+}
+
 export async function renderCard(post, renderTopic = true) {
   return (
     div({ class: 'card' },
       div({ class: 'card-thumbnail' },
         a({ href: post.path },
-          createOptimizedPicture(post.image, post.header),
+          createOptimizedPicture(validPostImage(post.image), post.header),
```

Comment:
I think simply using `new URL(post.image, window.origin).toString()` should work.

it should:
* if the URL is fully qualified keep it as is
* if the URL is not fully qualified, use `window.origin`

Reply:
This works :)